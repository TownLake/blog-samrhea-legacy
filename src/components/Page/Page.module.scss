@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.page {
  @include margin-bottom(2);

  .inner {
    @include padding(1, 0.75, 0);
  }

  .title {
    font-size: $typographic-base-font-size * 2.5;
    font-weight: 600;
    @include line-height(2);
    @include margin-top(0);
    @include margin-bottom(1.45);
  }

  .body {
    font-size: $typographic-base-font-size;
    @include line-height(1);
    @include margin(0, 0, 1);
  }
}

@include breakpoint-sm {
  .page {
    lost-column: 7/12;

    .inner {
      @include padding(1.25, 0.75, 0);
    }
  }
}

@include breakpoint-md {
  .page {
    lost-column: 2/3;

    .inner {
      @include padding(1.5, 1, 0);
    }
  }
}
