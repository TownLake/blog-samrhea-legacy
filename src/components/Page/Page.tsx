import React, { useEffect, useRef } from "react";

import type { Nullable } from "@/types";

import * as styles from "./Page.module.scss";

interface Props {
  title?: string;
  children: React.ReactNode;
}

const Page: React.FC<Props> = ({ title, children }: Props) => {
  const pageRef = useRef<Nullable<HTMLDivElement>>(null);

  useEffect(() => {
    if (pageRef.current) {
      pageRef.current.scrollIntoView();
    }
  }, []);

  return (
    <div ref={pageRef} className={styles.page}>
      <div className={styles.inner}>
        {title && <h1 className={styles.title}>{title}</h1>}
        <div className={styles.body}>{children}</div>
      </div>
    </div>
  );
};

export default Page;
