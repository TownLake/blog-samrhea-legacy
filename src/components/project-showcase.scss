.project-showcase {
    max-width: 1200px;
    margin: 0 auto;
    padding: 2rem;
  
    h1 {
      text-align: center;
      margin-bottom: 2rem;
      font-size: 2rem;
      font-weight: bold;
    }
  
    .project-grid {
      display: grid;
      grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
      gap: 2rem;
    }
  }
  
  // Dark mode styles
  @media (prefers-color-scheme: dark) {
    .project-showcase {
      background-color: #1a202c;
      color: #f7fafc;
    }
  }