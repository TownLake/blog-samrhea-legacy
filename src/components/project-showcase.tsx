.project-tile {
    position: relative;
    width: 100%;
    height: 0;
    padding-bottom: 75%; // 4:3 aspect ratio
    overflow: hidden;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
  
    &:hover {
      transform: translateY(-5px);
    }
  
    .project-link {
      text-decoration: none;
      color: inherit;
    }
  
    .project-image {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-size: cover;
      background-position: center;
    }
  
    .project-overlay {
      position: absolute;
      bottom: 0;
      left: 0;
      right: 0;
      background: rgba(0, 0, 0, 0.7);
      color: white;
      padding: 1rem;
      transform: translateY(100%);
      transition: transform 0.3s ease;
    }
  
    &:hover .project-overlay {
      transform: translateY(0);
    }
  
    .project-title {
      margin: 0 0 0.5rem;
      font-size: 1.2rem;
    }
  
    .project-description {
      margin: 0;
      font-size: 0.9rem;
      opacity: 0.8;
    }
  }