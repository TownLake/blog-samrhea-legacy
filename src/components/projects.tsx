import React from 'react';
import { Link } from 'gatsby';
import './project.scss';

interface ProjectProps {
  title: string;
  description: string;
  imageUrl: string;
  projectUrl: string;
}

const Project: React.FC<ProjectProps> = ({ title, description, imageUrl, projectUrl }) => {
  return (
    <div className="project-tile">
      <Link to={projectUrl} className="project-link">
        <div className="project-image" style={{ backgroundImage: `url(${imageUrl})` }}>
          <div className="project-overlay">
            <h3 className="project-title">{title}</h3>
            <p className="project-description">{description}</p>
          </div>
        </div>
      </Link>
    </div>
  );
};

export default Project;