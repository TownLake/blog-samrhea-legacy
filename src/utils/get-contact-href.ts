import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
