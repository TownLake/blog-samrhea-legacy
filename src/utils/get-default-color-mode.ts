const getDefaultColorMode = (): "dark" | "light" => {
  if (typeof window === "undefined") {
    return "light";
  }

  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "light";
};

export default getDefaultColorMode;
