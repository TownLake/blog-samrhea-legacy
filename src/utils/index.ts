export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
