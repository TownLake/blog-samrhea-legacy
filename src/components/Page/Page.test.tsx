import React from "react";

import { Page } from "@/components/Page";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Page", () => {
  test("renders correctly", () => {
    const props = {
      children: mocks.markdownRemark.html,
      title: mocks.markdownRemark.frontmatter.title,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Page {...props}>{props.children}</Page>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
