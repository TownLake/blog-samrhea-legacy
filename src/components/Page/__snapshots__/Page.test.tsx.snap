// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div>
  <div>
    <h1>
      Perfecting the Art of Perfection
    </h1>
    <div>
      &lt;p&gt;An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.&lt;/p&gt;
    </div>
  </div>
</div>
`;
